engine/runtime-integration-tests/src/test/java/org/enso/compiler/ErrorCompilerTest.java

```diff
@@ -505,6 +505,18 @@ public void illegalPrivateTypeDeclaration() throws Exception {
         ir, Syntax.UnexpectedExpression$.MODULE$, "Unexpected expression", 0, 14);
   }
 
+  // Conversion methods cannot be specified as private
+  @Test
+  public void illegalPrivateConversion() throws Exception {
+    var ir = parse("""
+        type My_Type
+        type Other_Type
+        private My_Type.from (other:Other_Type) = 42
+        """);
+    assertSingleSyntaxError(
+        ir, Syntax.UnexpectedExpression$.MODULE$, "Unexpected expression", 29, 73);
```

Comment:
Is there any way that we could get more descriptive error messages than `"Unexpected expression"`?